app/mailers/user_mailer.rb

```diff
@@ -8,68 +8,37 @@ class UserMailer < ActionMailer::Base
     default from: AppConfig.first.admin_email, cc: AppConfig.first.admin_email
   end
 
-  def checkin_receipt(reservation)
+  # checks the status of the current reservation and sends the appropriate email
+  # checkout_receipt forces a check out checkout_receipt to be sent
+  def reservation_status_update(reservation, checkout_receipt = false) # rubocop:disable all
     set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Your equipment return receipt')
-  end
-
-  def checkout_receipt(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Your equipment checkout receipt')
-  end
-
-  def missed_reservation_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation Missed')
-  end
-
-  def overdue_checkin_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] OVERDUE: equipment checkin')
-  end
 
-  def overdue_checked_in_fine(overdue_checked_in)
-    return if overdue_checked_in.equipment_model.late_fee == 0
-    set_app_config
-    @overdue_checked_in = overdue_checked_in
-    mail(to: overdue_checked_in.reserver.email,
-         subject: '[Reservations] Overdue equipment fine')
-  end
-
-  def reservation_confirmation(complete_reservation)
-    set_app_config
-    @complete_reservation = complete_reservation
-    mail(to: complete_reservation.first.reserver.email,
-         subject: '[Reservations] Confirmation of your reservation')
-  end
-
-  def upcoming_checkin_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reminder: equipment check in')
-  end
+    return if reservation.status == 'returned overdue' &&
+              reservation.equipment_model.late_fee == 0
 
-  def request_approved_notification(reservation)
-    set_app_config
-    @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation request approved!')
-  end
-
-  def request_denied_notification(reservation)
-    set_app_config
     @reservation = reservation
-    mail(to: reservation.reserver.email,
-         subject: '[Reservations] Reservation request denied')
+    @status = @reservation.status
+
+    if @reservation.start_date == Time.zone.today &&
+       @status == 'reserved'
+      @status = 'starts today'
+    elsif @reservation.due_date == Time.zone.today &&
+          @status == 'checked out'
+      @status = 'due today'
+    end
+
+    if checkout_receipt && (@status == 'due today' || @status == 'overdue') &&
+       @reservation.checked_out
```

Comment:
Hmm, I almost feel like we should `return` here if it's not checked out. As I see it:

1. If `checkout_receipt` (or `receipt`, I actually prefer the latter since we made one controller method for both cases)
  1. If checked out and status weirdness then fix status
  2. If checked out and status is checked out then do nothing
  3. If not checked out, don't send the e-mail
  4. If checked in, do nothing

Does that make sense?